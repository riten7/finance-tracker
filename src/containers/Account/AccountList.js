import React from 'react';
import { useSelector } from 'react-redux';

import { Collapse, Typography } from 'antd';

import { groupAccounts } from '../../utility/constant';
import AccountListItem from './AccountListItem';
import './account.css';

const { Panel } = Collapse;
const { Text } = Typography;

const NoAccounts = () => (
  <div className="accounts-form__empty">You don't have any accounts created yet.</div>
);

const AccountList = () => {
  const { accounts } = useSelector(state => state.accounts);
  const groups = groupAccounts(accounts);
  const groupKeys = Object.keys(groups);

  return (
    <>
      <h2>Accounts</h2>
      {groupKeys.length > 0 ?
        <Collapse>
          {groupKeys.map(key => (
            <Panel showArrow={true} header={groups[key].name} key={key} extra={<Text type={groups[key].total > 0 ? 'success' : 'danger'}>NPR. {groups[key].total}</Text>}>
              {groups[key].accounts.map(account => (
                <AccountListItem
                  key={account.id}
                  account={account}
                />
              ))}
            </Panel>
          ))}
        </Collapse> : <NoAccounts />
      }
    </>
  )
}

export default AccountList;
