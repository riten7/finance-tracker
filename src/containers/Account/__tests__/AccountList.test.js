import React from 'react';
import { mount } from "enzyme";
import { Provider } from 'react-redux';
import configureStore from "redux-mock-store";
import AccountList from '../AccountList';
import data from '../../../__mockData__/data.json';

describe("AccountList component", () => {
  let wrapper;
  const store = configureStore()(data);
   it("Component renders properly", () => {
    wrapper = mount(<Provider store={store}><AccountList /></Provider>);
    expect(wrapper.html()).toMatchSnapshot();
  });
});