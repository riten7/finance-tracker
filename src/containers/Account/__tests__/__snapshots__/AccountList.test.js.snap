// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountList component Component renders properly 1`] = `"<h2>Accounts</h2><div class=\\"ant-collapse ant-collapse-icon-position-left\\"><div class=\\"ant-collapse-item ant-collapse-no-arrow\\"><div class=\\"ant-collapse-header\\" role=\\"button\\" tabindex=\\"0\\" aria-expanded=\\"false\\">Cash<div class=\\"ant-collapse-extra\\">NPR. 131406817</div></div></div></div>"`;
